espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/Screenshot.java

```diff
@@ -0,0 +1,88 @@
+package io.appium.espressoserver.lib.handlers;
+
+import android.graphics.Bitmap;
+import android.os.Environment;
+import android.util.Base64;
+import android.view.View;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Date;
+
+
+import io.appium.espressoserver.lib.handlers.exceptions.AppiumException;
+import io.appium.espressoserver.lib.model.AppiumParams;
+import io.appium.espressoserver.lib.viewaction.RootViewFinder;
+
+public class Screenshot implements RequestHandler<AppiumParams, String> {
+
+    @Override
+    public String handle(AppiumParams params) throws AppiumException {
+        return takeScreenshot();
+    }
+
+
+    private String takeScreenshot() throws AppiumException {
+        Date now = new Date();
+        android.text.format.DateFormat.format("yyyy-MM-dd_hh:mm:ss", now);
+
+        try {
+            // image naming and path  to include sd card  appending name you choose for file
+            String mPath = Environment.getExternalStorageDirectory().toString() + "/" + now + ".jpg";
```

Comment:
in Java world it's more useful to use File type for paths and other stuff. This way one can define it as:

```java
File screenshot = new File(Environment.getExternalStorageDirectory().toString(), fileName + ".jpg")
```

path separator will be added automatically